addons/finalAlert/finalAlert.lua

```diff
@@ -0,0 +1,303 @@
+            end
+
+            settings:save()
+        elseif cmd == "pos" then
+            local x = tonumber(args[1])
+            local y = tonumber(args[2])
+
+            if type(x) == "number" and type(y) == "number" then
+                settings.x_position = x
+                settings.y_position = y
+                settings:save()
+                refresh_backgrounds()
+                print("Moved display to: " .. args[1] .. ", " .. args[2])
+            else
+                print("Please specify x and y coordinates.")
+            end
+        elseif cmd == "size" then
+            local size = args[1]
+
+            if size == "small" or size == "regular" then
+                settings.background_size = size
+                settings:save()
+                refresh_backgrounds()
+                print("Display size set to " .. size .. ".")
+            else
+                print('Please specify "small" or "regular" for the size.')
+            end
+        elseif cmd == "duration" then
+            local duration = tonumber(args[1])
+
+            if type(duration) == "number" and duration > 0 then
+                settings.trigger_duration = duration
+                print("Display duration set to " .. duration .. " secs.")
+            else
+                print("Please specify a positive number.")
+            end
+        elseif cmd == "sounds" then
+            local state = args[1]
+
+            if state == "on" or state == "off" then
+                settings.sounds = state
+                settings:save()
+                print("Sounds have been turned " .. state .. ".")
+            else
+                print('Please specify "on" or "off" for sounds.')
+            end
+        else
+            print("Unrecognized command.")
+        end
+    end
+)
+
+windower.register_event(
+    "action",
+    function(act)
+        local target
+        local t = windower.ffxi.get_mob_by_target("t")
+        local bt = windower.ffxi.get_mob_by_target("bt")
+
+        if t and t.is_npc and not t.in_party and not t.in_alliance then
+            target = t.id
+        elseif bt then
+            target = bt.id
+        else
+            return
+        end
+
+        if act.category == weapon_skill_category and act.actor_id == target then
+            local skill_name =
+                res.monster_abilities[act.targets[1].actions[1].param] and
+                res.monster_abilities[act.targets[1].actions[1].param].name or
+                "???"
+
+            if act.param == interrupt_id then
+                skill_name = "Interrupted!"
+                show_caption(skill_name, "int")
+            else
+                show_caption(skill_name, "ws")
+            end
+        elseif act.category == magic_category and act.actor_id == target then
+            local spell_name =
+                res.spells[act.targets[1].actions[1].param] and res.spells[act.targets[1].actions[1].param].name or
```

Comment:
It bothers me a bit that this is broken up into three lines in the WS block, but only two lines in the magic block :\ I'm starting to get the feeling you're using a fixed width and wrap lines exceeding it, in which case I'd vote to just leave them be. Fixed-width line wraps are never a good idea, despite what the Python style guide says :|